infra/src/main/kotlin/fi/fta/geoviite/infra/inframodel/InfraModelParsing.kt

```diff
@@ -176,14 +176,28 @@ val xmlCharsets = listOf(
     StandardCharsets.US_ASCII,
     StandardCharsets.ISO_8859_1,
 )
-fun getEncoding(bytes: ByteArray): Charset {
-    ByteArrayInputStream(bytes).use { stream ->
-        val xmlStreamReader: XMLStreamReader = XMLInputFactory.newInstance().createXMLStreamReader(stream)
-        val fileEncoding = xmlStreamReader.encoding
-        val encodingFromXMLDeclaration = xmlStreamReader.characterEncodingScheme
-        return (encodingFromXMLDeclaration ?: fileEncoding)?.let { name ->
+
+fun mapBomToCharset(bom: ByteOrderMark) =
```

Comment:
BOM:eja on muitakin, mutta tässä on tarkoituksella mäpätty vain XML:n hyväksymien UTF:ien BOM:it

Reply:
:+1: